### ISSUE TITLE
Not all the close neighbours are found until search radius is increased

### ISSUE THREAD

**Issue by pnuu** [pytroll/pyresample#113]
#### Code Sample, a minimal, complete, and verifiable piece of code

```python
#!/usr/bin/env python

import numpy as np
from pyresample import geometry
from pyresample import kd_tree
from satpy.resample import get_area_def
import warnings

# Target area, 4x4 pixels
target_def = geometry.AreaDefinition('areaD', 'Europe (3km, HRV, VTC)',
                                     'areaD',
                                     {'a': '6378144.0', 'b': '6356759.0',
                                      'lat_0': '50.00', 'lat_ts': '50.00',
                                      'lon_0': '8.00', 'proj': 'stere'},
                                     4, 4,
                                     [-1370912.72, -909968.64000000001,
                                      1029087.28, 1490031.3600000001])

# Source area
# These coordinates exceed the extent of target area in the north
lons, lats = np.meshgrid(np.linspace(-25., 40., num=100),
                         np.linspace(45., 75., num=100))
data = np.ones((100, 100))
source_def = geometry.SwathDefinition(lons=lons, lats=lats)

# 50 km, 500 km and 5000 km search radii
for radius in [50e3, 500e3, 5000e3]:
    # Hide the warning about "more than 32 neighbours" for larger radii
    with warnings.catch_warnings():
        warnings.simplefilter("ignore")
        (input_idxs, output_idxs, idx_ref, dists) = \
            kd_tree.get_neighbour_info(source_def, target_def,
                                       radius, neighbours=32,
                                       nprocs=1)
    print(dists[0, :6])

```
#### Problem description

In some cases, like the north-western location in this target area (pixel location [0, 0]), the search radius needs to be increased to unreasonable levels before the closest neighbours are retrieved. Reproducable with both pykdtree and scipy.spatial kd-tree implementations.

#### Expected Output

The four locations closer than 50 km distance should be found with all the search radii used in the above example.

#### Actual Result, Traceback if applicable

With increasing radius, more close neighbours are found:

```python
[ 22063.30382373  26983.85276005             inf             inf
             inf             inf]
[ 22063.30382373  26983.85276005  50161.67572158  56500.08927782
  56922.78772955  58391.07322077]
[ 22063.30382373  23653.54759765  26983.85276005  28287.49072995
  50161.67572158  50887.5203979 ]
```

#### Versions of Python, package at hand and relevant dependencies

PyResample develop, pyKDTree 1.2.2 (master branch from github). Might cause the failing test in #112

**Comment by pnuu** [comment]
Manual calculation of the distances for the same location ([0, 0]) of the target area.

```python

import numpy as np
from pyresample import geometry
from pyresample import _spatial_mp

# Converter from lonlats to cartesian coordinates
cartesian = _spatial_mp.Cartesian()

# Source data coordinates
lons, lats = np.meshgrid(np.linspace(-25., 40., num=100), np.linspace(45., 75., num=100))
# Source coordinates in cartesian space
input_coords = cartesian.transform_lonlats(lons.ravel(), lats.ravel())

# Target area definition
target_def = geometry.AreaDefinition('areaD', 'Europe (3km, HRV, VTC)',
                                     'areaD',
                                     {'a': '6378144.0', 'b': '6356759.0',
                                      'lat_0': '50.00', 'lat_ts': '50.00',
                                      'lon_0': '8.00', 'proj': 'stere'},
                                     4, 4,
                                     [-1370912.72, -909968.64000000001,
                                      1029087.28, 1490031.3600000001])
# Target coordinates
t_lons, t_lats = target_def.get_lonlats()
# Target coordinates in cartesian space
output_coords = cartesian.transform_lonlats(t_lons.ravel(), t_lats.ravel())

# Calculate distance for the north-western corner point
a_d = (input_coords[:, 0] - output_coords[0, 0])**2
b_d = (input_coords[:, 1] - output_coords[0, 1])**2
c_d = (input_coords[:, 2] - output_coords[0, 2])**2
dists = np.sqrt(a_d + b_d + c_d)

# Print the 6 shortest distances
print(np.sort(dists)[:6])
```

This prints the same values as when using 5000 km search radius above:

```python
[ 22063.30382373  23653.54759765  26983.85276005  28287.49072995
  50161.67572158  50887.5203979 ]
```

**Comment by pnuu** [comment]
And continuing from the previous snippet using the `input_coords` and `output_coords` variables:

```python
from pykdtree.kdtree import KDTree

resample_kdtree = KDTree(input_coords)

for radius in [50e3, 500e3, 5000e3]:
   distance_array, index_array = resample_kdtree.query(output_coords,
                                                       k=32,
                                                       distance_upper_bound=radius)
   print distance_array[0, :6]
```

we get:

```python
[ 22063.30382373  23653.54759765  26983.85276005  28287.49072995
             inf             inf]
[ 22063.30382373  23653.54759765  26983.85276005  28287.49072995
  50161.67572158  50887.5203979 ]
[ 22063.30382373  23653.54759765  26983.85276005  28287.49072995
  50161.67572158  50887.5203979 ]
```

In this case the results are consistent! So we've now isolated the problem to pyresample.

**Comment by pnuu** [comment]
Seems to be solved by setting `reduce_data=False` to the `get_neighbour_info(...)` call.

**Comment by pnuu** [comment]
The problem was in `data_reduce._get_valid_index()` where Earth-centric angle was used to extend the boundaries. The error was that the angle was not converted to degrees before subtracting from the longitudes and/or latitudes. Fixed by #112 .
